reader.go

```diff
@@ -740,44 +722,37 @@ func (r *Reader) commitLoopInterval(conn offsetCommitter, stop <-chan struct{})
 	ticker := time.NewTicker(r.config.HeartbeatInterval)
 	defer ticker.Stop()
 
-	defer func() {
-		// commits any outstanding offsets on close
-		if err := r.commitOffsetsWithRetry(conn, r.offsetStash, defaultCommitRetries); err == nil {
-			r.offsetStash.reset()
+	commit := func() {
+		if err := r.commitOffsetStash(conn); err != nil {
+			r.withErrorLogger(func(l *log.Logger) {
+				l.Print(err)
+			})
 		}
-	}()
+	}
 
 	for {
 		select {
 		case <-stop:
+			commit()
 			return
 
 		case <-ticker.C:
-			if len(r.offsetStash) == 0 {
-				continue
-			}
+			commit()
 
-			if err := r.commitOffsetsWithRetry(conn, r.offsetStash, defaultCommitRetries); err != nil {
-				r.withErrorLogger(func(l *log.Logger) {
-					l.Printf("unable to commit offset: %v", err)
-				})
-				return
-			}
-			r.offsetStash.reset()
-
-		case msgs, ok := <-r.commits:
-			if !ok {
-				r.withErrorLogger(func(l *log.Logger) {
-					l.Println("reader commit channel unexpectedly closed")
-				})
-				return
-			}
-
-			r.offsetStash.merge(msgs...)
+		case req := <-r.commits:
+			r.offsetStash.merge(req.commits)
 		}
 	}
 }
 
+func (r *Reader) commitOffsetStash(conn offsetCommitter) error {
```

Comment:
given commitOffsetStash is only called from one place and it only performs a single task and is already wrapped in the commit func, is there a reason not to inline it above?

Reply:
Not really, I can change that 👍